### ISSUE TITLE
Poor WMS performance on bigger screens 

### ISSUE THREAD

**Issue by joaoces19** [openlayers/openlayers#7790]
Hi, first of all thanks for your work in such an awesome opensource library like OpenLayers.

I would like to report poor map performance in bigger resolutions loading multiple WMS layers. 

The example shown in the video loads 14 sample WMS layers from GeoServer and the map performance decreases significantly when you drag and zoom on a bigger screen. The video attached shows that, first on a smaller screen with a 1366x768 resolution and second (from 18 seconds) with a bigger screen with a 1920x1080 resolution. 

CodePen: https://codepen.io/joaoces/pen/rJjMNj

Video: https://www.dropbox.com/s/2h9u6xpxuw7be9y/ol4_bad_performanc_multiple_layers.mp4?dl=0

Computer specs: 
- Windows 10 64 bits
- Chrome Version 63.0.3239.132 (Official compilation) (64 bits)
- Intel Core i5-4200U @ 2.3GHz
- 16 GB of RAM
- ADATA SP900 250 GB SDD

Thanks
João César

**Comment by ahocevar** [comment]
You're putting unnecessary load on both your server, network connection and your browser's renderer. There are two ways to optimize:

 1. Configure the source with `ratio: 1` to only request and process the extent that is currently shown on the map.
 2. Combine WMS layers from the same WMS into a single `ol.layer.Image` instance.

So if you now have something like
```js
new ol.Map({
  layers: [new ol.layer.Image({
    source: new ol.source.ImageWMS({
      url: url,
      params: [
        LAYERS: 'layer1'
      ]
   })
  }), new ol.layer.Image({
    source: new ol.source.ImageWMS({
      url: url,
      params: [
        LAYERS: 'layer2'
      ]
   })
  })]
})
```
you should have the following instead:
```js
new ol.Map({
  layers: [new ol.layer.Image({
    source: new ol.source.ImageWMS({
      ratio: 1,
      url: url,
      params: [
        LAYERS: 'layer1, layer2'
      ]
   })
  })]
});
```
Also note that you should not pass an `SRS` (or `CRS`) param to the WMS Source. That is set automatically. If you have a WMS that does not support your view projection, you have to set `projection` on the WMS source instead, which will trigger raster reprojection, which adds another potential performance problem. So it is always better to use the view projection if provided by the WMS.

**Comment by tixvieira** [comment]
Hi @ahocevar ,

Me and @joaoces19,

At start, we think that the browsers at this time already can handle a lot of layers on a full screen resolution like 1920x1080. But ok, there's always some limitations. :)

We have a lot of layers, and a big layerswitcher, so we've developed and extension for the layerswitcher to combine all the layers dynamically into some groups that we define. We've already tested by changing the default ratio (1.5) to 1, and it helps a bit, but once more depending of the resolution.

For now, we got a good behavior.

Thank you,
Tiago Vieira
